lib/cn/cn_tree.c

```diff
@@ -923,12 +926,13 @@ cn_comp_release(struct cn_compaction_work *w)
         }
 
         /* If cn_comp_update_XXX() was called and was successful then it will
-         * have updated tn_split_cnt, busycnt, and the token.  For all errors
-         * we must do it here.
+         * have updated tn_ss_splitting, tn_busycnt, and the token.  For all
+         * errors we must do it here.
          */
         if (w->cw_action == CN_ACTION_SPLIT) {
-            w->cw_tree->ct_root->tn_split_cnt--;
-            w->cw_node->tn_split_cnt = 0;
+            w->cw_node->tn_ss_splitting = false;
+            w->cw_tree->ct_split_cnt--;
+            cn_subspill_wakeup(w->cw_node);
```

Comment:
You could put these three lines in a static function instead of duplicating the code.

Reply:
Then you wouldn't even need the comment.

Reply:
Hmm, lemme see if I can move all this stuff from both places into cn_comp_cleanup()...

Reply:
Seems like we need only one of cn_comp_cleanup() or cn_comp_release()....

Reply:
That seems to be working...